src/request_handler.rs

```diff
@@ -50,7 +58,9 @@ impl RequestHandler {
         }
         body {
           @while let Some(entry) = read_dir.next_entry().await? {
-            (entry.file_name().to_string_lossy())
+            a download href=(format!("./{}", entry.file_name().to_string_lossy())) {
```

Comment:
This will have weird behavior for non-unicode file names. They'll appear in listings, but you'll get a 404 when you click them, because the client will try to retrieve them with a path that includes the unicode replacement character (�) instead of the actual bytes in the path.

I think what we should probably do right now is just not list files with non-unicode paths, and somehow warn the user about the situation.

We could somehow generate links that contained a representation of the path (like hex) and decoded it when the file was retrieved. However, I think in general it's probably best to just refuse to handle non-unicode text, and force the user to deal with it, since it avoids a ton of complexity. If users want non-unicode filenames, we could eventually support this.

If we encounter a non-unicode path, we could:

1. Return an error
2. Print a warning to the console
3. Display something like "N files with non-unicode paths omitted" in the directory listing

1 seems extreme, 2 seems reasonable but users might not look at the console and wonder where their non-unicode files are, and 3 is more obvious but would bother users who don't care and just want to see the files that they can get. I'm leaning towards 2. What do you think?

Reply:
I vote for 2. and for deferring this to a future PR. See also #25.